### PR TITLE
chore(TS): Move object classes

### PR DESCRIPTION
**Pull request by ShaMan123**
## Description

Move all object proto to `shapes/Object` and rename them (since it doesn't make sense to have them under mixins anymore).
I would argue that the mixins folder should go away

## Changes

- extracted `resolveOrigin` logic from object origin to misc, keeping all the object proto "hidden"/"private" accept `FabricObject` and `Object`
- the rest are imports

### CODE REVIEW

**Review by ShaMan123 — COMMENTED** [review]
ready
will conflict at least with #8404 but it is limited in scope so porting changes is fine

File: src/shapes/polygon.class.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { fabric } from '../../HEADER';
 import { TClassProperties } from '../typedefs';
-import { FabricObject } from './object.class';
+import { FabricObject } from './Object/FabricObject';
```
[the remark this answers is not in the document]
Comment (asturur) [reply src/shapes/polygon.class.ts:3]: oh static methods are inherited too?
That is ok, let's keep this PR to import shuffling only

**Comment by asturur** [comment]
I think i m ok with this PR, i do see a loss in moving files around and losing years of position memory. But i think i ll survive

**Review by asturur — APPROVED** [review]

File: src/shapes/polygon.class.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (ShaMan123) [reply src/shapes/polygon.class.ts:3]: Yep
I was surprised
Great

**Comment by ShaMan123** [comment]
> I think i m ok with this PR, i do see a loss in moving files around and losing years of position memory. But i think i ll survive

You mean with the blame?

**Pull request merged**